test/test_python_dispatch.py

```diff
@@ -105,10 +105,6 @@ def test_list_ret(self) -> None:
         # test all sequence types are permissible returns
         for list_type in (list, tuple):
             class A(torch._C._TensorBase):
-                @staticmethod
-                def __new__(cls, elem):
-                    return torch.Tensor._make_subclass(cls, elem, elem.requires_grad)
```

Comment:
How does the new constructor work with wrapper Tensors? I guess I want to see an example of how this changes construction for the different type of tensor subclasses 

Someone might want to create a wrapper Tensor subclass (DiagTensor?) that holds a vector and would want DiagTensor(blah) to work. Does this mean DiagTensor would override __new__ ?

Reply:
WrapperTensor still has to define __new__; in general any tensor which doesn't match metadata doesn't work here.